plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -277,16 +277,16 @@ function OPDSBrowser:fetchFeed(item_url, headers_only)
     local code, headers, status = socket.skip(1, http.request(request))
     socketutil:reset_timeout()
 
-    if headers == nil then
-        error(status or code or "network unreachable")
-    end
     if headers_only then
-        return headers["last-modified"]
+        return headers and headers["last-modified"]
     end
     if code == 200 then
         local xml = table.concat(sink)
         return xml ~= "" and xml
     end
+    if headers == nil then
+        error(status or code or "network unreachable")
+    end
```

Comment:
Shouldn't this be moved a bit further down below the other status codes?

Reply:
`headers.location` is used in the error messages.

Reply:
That applies only to 3xx.